### PR TITLE
[CPM-142] Move data between years

### PR DESCRIPTION
**Pull request by daltonlim**
## Changes
- Add drag and drop between years

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

Addresses [CPM-142]

Connects #y

[CPM-142]: https://momentum-agile.atlassian.net/browse/CPM-142

### CODE REVIEW

File: client/src/pages/Plan/Plan.js [hunk 1]
```diff
@@ -16,6 +16,28 @@ const Plan = () => {
     const { student, realCourses, programme } = usePlan();
     const [searchTerm, setSearchTerm] = useState("");
     const [note, setNote] = useState("");
+    const start = 2020;
+    const end = 2021;
+    const years = [];
+    for (var i = start; i <= end; i++) {
+        years.push(i);
+    }
+    console.log(years);
```
Comment (FocalChord) [inline client/src/pages/Plan/Plan.js:25]: surely u can make this `const years = [2020, 2021]` lol

File: client/src/pages/Plan/Plan.js [hunk 2]
```diff
@@ -16,6 +16,28 @@ const Plan = () => {
...
+        { course: "Softeng 255", year: 2020, semester: "S2" },
+        { course: "Softeng 256", year: 2020, semester: "S2" },
+        { course: "Softeng 351", year: 2021, semester: "S1" },
+        { course: "Softeng 352", year: 2021, semester: "S1" },
+        { course: "Softeng 353", year: 2021, semester: "S1" },
+        { course: "Softeng 354", year: 2021, semester: "S1" },
+        { course: "Softeng 451", year: 2021, semester: "S2" },
+        { course: "Softeng 452", year: 2021, semester: "S2" },
+        { course: "Softeng 453", year: 2021, semester: "S2" },
+    ]);
```
Comment (FocalChord) [inline client/src/pages/Plan/Plan.js:40]: is there  a backend for this or are u gonna do that on the other ticket

File: client/src/pages/Plan/Plan.js (hunk 1 above)
Comment (daltonlim) [reply client/src/pages/Plan/Plan.js:25]: A plan has a start and end year which this should represent

File: client/src/pages/Plan/Plan.js (hunk 2 above)
Comment (daltonlim) [reply client/src/pages/Plan/Plan.js:40]: Yea already part of a plan object

File: client/src/pages/Plan/ItemTypes.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 export const ItemTypes = {
     COURSE_PILL: "coursePill",
+    COURSE_TITLE: "courseTitle",
```
Comment (FocalChord) [inline client/src/pages/Plan/ItemTypes.js:3]: Should this be named `COURSE_TILE` instead of title

File: client/src/pages/Plan/Plan.js (hunk 1 above)
Comment (FocalChord) [reply client/src/pages/Plan/Plan.js:25]: Okay that's fine, can we get rid of console.log

Comment (daltonlim) [reply client/src/pages/Plan/Plan.js]: dooooooooone

File: client/src/pages/Plan/ItemTypes.js (hunk 3 above)
Comment (daltonlim) [reply client/src/pages/Plan/ItemTypes.js]: dun

**Review by FocalChord — APPROVED** [review]
okay sweet, seems to working well

**Review by anian100 — APPROVED** [review]
LGTM

**Pull request merged**